content/courses/Ada_For_The_C_Embedded_Developer/chapters/08_Performance.rst

```diff
@@ -37,10 +37,209 @@ Switches and optimizations
 Checks and assertions
 ---------------------
 
-.. todo::
+Checks
+~~~~~~
+
+Ada provides many runtime checks to ensure that the implementation is working
+as expected. For example, when accessing an array, we would like to make sure
+that we're not accessing a memory position that is not allocated for that
+array. This is achieved by an index check. Another example is the use of
+constrained ranges in numeric applications: when adding two integer numbers, we
```

Comment:
I would argue that the purpose of numeric ranges is to express some application-specific constraint that we want the compiler to help us enforce, and more importantly, that we want the compiler to tell us cannot be met when the underlying hardware cannot support the range given. In other words, there are two things to consider: just a range constraint, such as "A : Integer range 0 .. 10;" but also "type Result is range 0 .. 1_000_000_000;" Both represent some sort of application-specific constraint, but in addition, the type declaration promotes portability because it won't compile on targets that do not have a sufficiently large hardware numeric type. That's a definition of portability that is preferable to having something compile anywhere but not run correctly, as in C. But in any case overflow/underflow is a separate issue IMO.

Reply:
In this paragraph, I just wanted to give a simple example of overflow checking, but not discuss the purpose of range checking. I tried and rephrased it to make this clearer.

That being said, your argument is **very important**, but I don't think it belongs to a chapter about performance and optimizations. It fits, however, perfectly in the section about type ranges from chapter 2 ("The C Developer's Perspective on Ada"), so I included your comments (almost verbatim!) there. Thanks!